### ISSUE TITLE
a standard way for batch-creation of passwords/notes?

### ISSUE THREAD

**Issue by bogorad** [doy/rbw#220]
I needed to batch-create a couple of keys, so I found a hacky way to do it (spawn a fake "editor" that echos some stuff into the file provided as argument, use it as EDITOR=, then immediately delete it), but - maybe - there should be a less hacky way? :)

**Comment by Anachron** [comment]
Yep I need to migrate a ton of passwords and an option to disable the editor and just use whatever is on stdin would be very appreciated.

**Comment by doy** [comment]
yeah, overriding `EDITOR` should work fine currently, but i'd be open to a pr adding a flag to accept the password to set on stdin.

**Comment by anarcat** [comment]
what should be the format of that stdin anyways? it seems like everything gets stored as a note, but it would be nice if we could access the more complicated "structure" bitwarden passwords have, like fields and so on... we already have --uri on the commandline, perhaps there could be some way to have a similar syntax for other fields?

**Comment by bogorad** [comment]
> what should be the format of that stdin anyways

first line => password, the rest => note.

**Comment by anarcat** [comment]
> first line => password, the rest => note.

yeah, but that doesn't give me access to create new fields like "email".

the context here is i'm migrating from password-store and i would kind of like to have a way to have those (rather informal) key: value pairs converted into actual fields in vaultwarden...

stuff like OTP passwords would be nice to convert automatically as well: none of those are accessible from the commandline right now, from what i understand.

also, i just found a neat trick that one can use to avoid the EDITOR hack: just create the password with EDITOR=true, which will make an empty password, then pass the stuff through stdin to `rbw edit` which *does* support stdin!

**Comment by multi-io** [comment]
My context for this would be creating and storing AWS and Openstack app credentials, and you want to be able to have a "username" attribute that's separate from the name of the login entry. This isn't possible with rbw currently afaics, so I think this feature is desperately needed.

**Comment by varac** [comment]
Related: #246
